sw_launch_event.md

```diff
@@ -104,40 +100,27 @@ Notes:
 }
 ```
-Notes:
-* The `Request` that led to the navigation is included. In addition to the URL, this allows sites to inspect things like POST data (e.g. for a Web Share Target) that led to the invocation of the app. 
+
 
 ## Design Questions/Details
 
+### Should we expose the full `Request` instead of `LaunchParams` in the `LaunchEvent`
+
+For example for web share target, the data being shared could be part of the "POST" data of the request. By exposing a `Request` instead of (or in addition to) the `LaunchParams` this could be better handled by the service worker. However at the time launch handling runs, the request hasn't been created yet, so we'd have to create a synthetic request to be able to pass it in. Additionally if we do want to expose the POST data, we'd also want to expose that to `focus-existing` launch handlers, and thus would to add it to `LaunchParams` anyway.
+
 ### Restricting launch events to installed websites
 
-It is recommended that user agents only fire launch events for installed web apps. There are 2 reasons for this:
+By only triggering `launch` events when the manifest specifies the `service-worker` `client_mode`, we limit this functionality to installed web apps.
+
+There are 2 reasons for this:
  1. It is difficult to attribute bad behavior to misbehaving websites if they aren't installed (see the section below).
  2. It could be confusing if the behavior of clicking a link changes just because a user has visited a site that registered itself as a `launch` event handler. 
 
 Allowing launch events to be handled on the drive-by web could be explored in the future.
 
-### Handling redirects which trigger launch events
-
-We should allow launch events to be intercepted when a navigation triggers a redirect. For example imagine that https://www.example.com registers a `launch` event handler. The user then clicks a link to https://example.com. No handler is registered, so the navigation is processed. The navigation redirects to https://www.example.com. At this point, before the request is actually made, a launch event should be fired on the SW for https://www.example.com.
-
-If we don't do this, then many link clicks won't be correctly intercepted.
-
-When a `launch` event is captured as the result of a redirect, the behavior of the underlying page needs to be decided. User agents should have freedom to experiment here, but some options are:
-- The tab could be closed. This would be good behavior if a tab was only opened to process the navigation before it could be captured by the app.
-- The tab could return to the page it was on prior to the navigation starting. This seems like sensible behavior in the general case, allowing the user to continue on the site they were on prior to clicking the link handled by the launch event. The user may lose state on the site as a result of a navigation happening, but this wouldn't be unexpected since they've clicked a link.
-
-### Handling `window.open()`
```

Comment:
Are we removing this because for all cases that we COULD handle window.open, they all are `noopener` and thus don't return a frame reference?

Reply:
Not really, I'm mostly removing this because this entire web app launch handling spec merely describes what happens when a launch does happen, not what should or should not be considered a launch. So this seemed off-topic.

Reply:
ACK